File: abc/abc219/b.cpp
// https://atcoder.jp/contests/abc219/submissions/25934756

#include <bits/stdc++.h>
using namespace std;
int main() {
  string s, t, u;
  cin >> s >> t >> u;
  char T;
  while(cin >> T) {
    if(T == '1') cout << s;
    else if(T == '2') cout << t;
    else cout << u;
  }
}

File: abc/abc219/b.java
// https://atcoder.jp/contests/abc219/submissions/43260095

import java.util.Scanner;

class Main {
  public static void main(String[] args) {
    Scanner stdIn = new Scanner(System.in);

    String[] s = new String[3];
    for(int i = 0; i < 3; i++) s[i] = stdIn.next();
    String t = stdIn.next();
    int n = t.length();

    for(int i = 0; i < n; i++) {
      int idx = t.charAt(i) - 49;
      System.out.print(s[idx]);
    }
  }
}
